docs/src/cookbooks/naming.md

```diff
@@ -72,12 +72,91 @@ ChiselStage.emitSystemVerilog(new ExampleWhenPrefix)
 `_GEN` signals are usually generated from the FIRRTL compiler, rather than the Chisel library. We are working on
 renaming these signals with more context-dependent names, but it is a work in progress. Thanks for caring!
 
-### My module names are super unstable - I change one thing and Queue_1 becomes Queue_42. Help!
+### My module names are super unstable - I change one thing and Module_1 becomes Module_42. Help!
 
-This is the infamous `Queue` instability problem. In general, these cases are best solved at the source - the module
-itself! If you overwrite `desiredName` to include parameter information (see the
-[explanation](../explanations/naming#set-a-module-name) for more info), then this can avoid this problem permanantly.
-We've done this with some Chisel utilities with great results!
+This is an example of the module instability problem, which results from several modules all sharing the exact same name. To fix this, you must add more specificity to your `Module`'s name to avoid these name collisions.
+
+This can be done by leveraging the `desiredName` and `typeName` APIs. 
+`desiredName` is for indicating the names of `Modules` (e.g. influenced by the parameters passed in), and `typeName` is useful for modules which are type-parameterized by subclasses of `Data`. Overriding `desiredName` can reduce or even eliminate name collisions. For instance, suppose your module looks like the following:
+
+```scala mdoc:silent
+class MyModule[T <: Data](gen: T) extends Module {
+  // ...
+}
+```
+
+We can override `desiredName` of the module to include the type name of the `gen` parameter like so:
+
+```scala mdoc:silent:reset
+import chisel3._
+
+class MyModule[T <: Data](gen: T) extends Module {
+  override def desiredName = s"MyModule_${gen.typeName}"
+}
+```
+
+Any instances of your `MyModule` will now have Verilog module names containing the type parameter.
+
+```scala mdoc:compile-only
+val foo = Module(new MyModule(UInt(4.W))) // MyModule_UInt4
+val bar = Module(new MyModule(Vec(3, UInt(4.W)))) // MyModule_Vec3_UInt4
+```
+
+### Does the `typeName` apply for built-in module names too, most notably `Queue`?
```

Comment:
Sinc this isn't really a "recipe", I'd probably just move all this underneath the next topic maybe after the quick 'recipe' answer as a deeper explanation